### DISCUSSION TITLE
Control of librespot source?

### DISCUSSION THREAD

**Discussion by Moudilu** [snapcast/snapcast#1204] [Q&A]
Hi there

I have recently setup Snapcast (really love the project ❤️) with Mopidy and the native librespot as sources. I can control (play/pause) Mopidy through the Iris Webinterface of Mopidy and through Snapcast Web (I installed the react branch of it), however, I can't figure out how to control librespot.

Is this expected behaviour or did I do something wrong? Is there a controlscript for it (I tried meta_librespot-java.py without luck, Snapcast logs gave me an error about the socket already being open)? Did I maybe install Snapserver the wrong way/from the wrong source (see below for my installed version & source)?   
Or is there another way to easily achieve a controllable Spotify source?

Thanks for all the inputs!
Best, Fabian

My sources configuration in `/etc/snapserver.conf`:
```
...
source = alsa:///?name=Mopidy&device=hw:0,1,0&controlscript=meta_mopidy.py
source = librespot:///usr/bin/librespot?name=Spotify Multiroom&devicename=Multiroom
...
```

The librespot binary on my machine has been installed through installing Raspotify.
```
$ librespot --version
librespot 0.4.2 a6e1258 (Built on 2023-06-21, Build ID: g9ifMDjd, Profile: release)
```

I installed Snapserver from the almost latest package build (snapcast_arm64-debian-bookworm-8d282c557949f46c8b7f701381a3e50e30b4716d from https://github.com/badaix/snapcast/actions/runs/7854770807), since the released v0.27.0 doesn't install on my Raspberry Pi 4 with Debian Bookworm. 
```
$ snapserver --version
snapserver v0.27.0 (rev 8d282c55)
Copyright (C) 2014-2022 BadAix (snapcast@badaix.de).
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Johannes M. Pohl and contributors <https://github.com/badaix/snapcast/graphs/contributors>.
```

**Comment by StuartIanNaylor** [comment]
Snapserver & Snapclient are individual packages in RaspiOS Debian Bookworm as with a doh of thinking it didn't have snapcast I found the 2 seperate debs.
Dunno about Librespot.

**Comment by Moudilu** [comment]
Thanks for pointing that out, I thought it has only Snapcast 0.26 in the repository, in the backport repository there is the 0.27 version though it seems, that makes installation easier. However that doesn't solve that librespot isn't controllable. Or can you control it through the Snapcast interface in your setup? I get the opinion that this is currently not possible...

**Comment by StuartIanNaylor** [comment]
Haven't work out the plugins, but haven't tried as I think librespot needs a prem account, which I don't have.
The only thing I could hellp with was to say Snapserver and Snapclient seem to be seperate packages on Debian.
